gstreamer-plugin/git-patch/0001-svthevcenc-Add-svthevc-encoder-element.patch

```diff
@@ -0,0 +1,2123 @@
+From 19f59d2c7fff74a8828231ebdc1a4196dfdd3d50 Mon Sep 17 00:00:00 2001
+From: Yeongjin Jeong <yeongjin.jeong@navercorp.com>
+Date: Sat, 23 Mar 2019 21:01:51 +0900
+Subject: [PATCH] svthevcenc: Add svthevc encoder element
+
+---
+ configure.ac                      |    8 +
+ ext/Makefile.am                   |   12 +-
+ ext/meson.build                   |    1 +
+ ext/svthevc/Makefile.am           |   18 +
+ ext/svthevc/gstsvthevcenc.c       | 1561 +++++++++++++++++++++++++++++++++++++
+ ext/svthevc/gstsvthevcenc.h       |  118 +++
+ ext/svthevc/meson.build           |   16 +
+ tests/check/Makefile.am           |    7 +
+ tests/check/elements/.gitignore   |    1 +
+ tests/check/elements/svthevcenc.c |  228 ++++++
+ tests/check/meson.build           |    1 +
+ 11 files changed, 1969 insertions(+), 2 deletions(-)
+ create mode 100644 ext/svthevc/Makefile.am
+ create mode 100644 ext/svthevc/gstsvthevcenc.c
+ create mode 100644 ext/svthevc/gstsvthevcenc.h
+ create mode 100644 ext/svthevc/meson.build
+ create mode 100644 tests/check/elements/svthevcenc.c
+
+diff --git a/configure.ac b/configure.ac
+index 943412c..c8714e7 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -2310,6 +2310,12 @@ AG_GST_CHECK_FEATURE(WEBRTC, [WebRTC], webrtc, [
+   ])
+ ])
+ 
++dnl *** svthevc : works with libSVT-HEVC 1.3.0 ***
++translit(dnm, m, l) AM_CONDITIONAL(USE_SVTHEVC, true)
++AG_GST_CHECK_FEATURE(SVTHEVC, [svthevc plug-in], svthevc, [
++  AG_GST_PKG_CHECK_MODULES(SVTHEVC, SvtHevcEnc >= 1.3.0)
++])
++
+ else
+ 
+ dnl not building plugins with external dependencies,
+@@ -2393,6 +2399,7 @@ AM_CONDITIONAL(USE_X265, false)
+ AM_CONDITIONAL(USE_DTLS, false)
+ AM_CONDITIONAL(USE_VULKAN, false)
+ AM_CONDITIONAL(USE_TTML, false)
++AM_CONDITIONAL(USE_SVTHEVC, false)
+ 
+ fi dnl of EXT plugins
+ 
+@@ -2686,6 +2693,7 @@ ext/dtls/Makefile
+ ext/webrtc/Makefile
+ ext/webrtcdsp/Makefile
+ ext/ttml/Makefile
++ext/svthevc/Makefile
+ po/Makefile.in
+ docs/Makefile
+ docs/plugins/Makefile
+diff --git a/ext/Makefile.am b/ext/Makefile.am
+index 4de9214..d3ed2bb 100644
+--- a/ext/Makefile.am
++++ b/ext/Makefile.am
+@@ -400,6 +400,12 @@ else
+ TTML_DIR=
+ endif
+ 
++if USE_SVTHEVC
++SVTHEVC_DIR=svthevc
++else
++SVTHEVC_DIR=
++endif
++
+ SUBDIRS=\
+ 	$(VOAACENC_DIR) \
+ 	$(ASSRENDER_DIR) \
+@@ -468,7 +474,8 @@ SUBDIRS=\
+ 	$(VULKAN_DIR) \
+ 	$(WEBRTCDSP_DIR) \
+ 	$(TTML_DIR) \
+-	$(WEBRTC_DIR)
++	$(WEBRTC_DIR) \
++	$(SVTHEVC_DIR)
+ 
+ DIST_SUBDIRS = \
+ 	assrender \
+@@ -535,6 +542,7 @@ DIST_SUBDIRS = \
+ 	vulkan \
+ 	webrtcdsp \
+ 	ttml \
+-	webrtc
++	webrtc \
++	svthevc
+ 
+ include $(top_srcdir)/common/parallel-subdirs.mak
+diff --git a/ext/meson.build b/ext/meson.build
+index cd193b7..918980d 100644
+--- a/ext/meson.build
++++ b/ext/meson.build
+@@ -67,3 +67,4 @@ subdir('webrtc')
+ subdir('webp')
+ subdir('x265')
+ subdir('zbar')
++subdir('svthevc')
+diff --git a/ext/svthevc/Makefile.am b/ext/svthevc/Makefile.am
```

Comment:
How about svthevcenc instead? In case there will be svthevcdec later.

Reply:
Ok, but when I finished work, can I squash that commit and initial commit after your review?

Reply:
I fixed this too.